// miyojs-filter-variables/variables.js
// Generated by CoffeeScript 1.8.0

/* (C) 2014 Narazaka : Licensed under The MIT License - http://narazaka.net/license/MIT?2014 */
var MiyoFilters;

if (typeof MiyoFilters === "undefined" || MiyoFilters === null) {
  MiyoFilters = {};
}

MiyoFilters.variables_initialize = {
  type: 'through',
  filter: function(argument, request, id, stash) {
    this.variables = {};
    this.variables_temporary = {};
    this.variables_load = (function(_this) {
      return function(file) {
        var file_path, fs, json_str, path;
        fs = require('fs');
        path = require('path');
        file_path = path.join(process.cwd(), file);
        try {
          json_str = fs.readFileSync(file_path, 'utf8');
          return _this.variables = JSON.parse(json_str);
        } catch (_error) {

        }
      };
    })(this);
    this.variables_save = (function(_this) {
      return function(file) {
        var file_path, fs, json_str, path;
        fs = require('fs');
        path = require('path');
        file_path = path.join(process.cwd(), file);
        json_str = JSON.stringify(_this.variables);
        return fs.writeFileSync(file_path, json_str, 'utf8');
      };
    })(this);
    if (this.filters.miyo_template_stash != null) {
      this.filters.miyo_template_stash.v = function(value, request, id) {
        return this.variables;
      };
      this.filters.miyo_template_stash.vt = function(value, request, id) {
        return this.variables_temporary;
      };
    }
    return argument;
  }
};

MiyoFilters.variables_load = {
  type: 'through',
  filter: function(argument, request, id, stash) {
    if ((argument != null ? argument.variables_load : void 0) == null) {
      throw 'argument.variables_load undefined';
    }
    this.variables_load(argument.variables_load);
    return argument;
  }
};

MiyoFilters.variables_save = {
  type: 'through',
  filter: function(argument, request, id, stash) {
    if ((argument != null ? argument.variables_save : void 0) == null) {
      throw 'argument.variables_save undefined';
    }
    this.variables_save(argument.variables_save);
    return argument;
  }
};

MiyoFilters.variables_set = {
  type: 'through',
  filter: function(argument, request, id, stash) {
    var name, pname;
    if ((argument != null ? argument.variables_set : void 0) == null) {
      throw 'argument.variables_set undefined';
    }
    for (name in argument.variables_set) {
      pname = name.replace(/\.[^.]+$/);
      this.variables[name] = this.property(argument.variables_set, pname, request, id, stash);
    }
    return argument;
  }
};

MiyoFilters.variables_delete = {
  type: 'through',
  filter: function(argument, request, id, stash) {
    var name, _i, _len, _ref;
    if ((argument != null ? argument.variables_delete : void 0) == null) {
      throw 'argument.variables_delete undefined';
    }
    _ref = argument.variables_delete;
    for (_i = 0, _len = _ref.length; _i < _len; _i++) {
      name = _ref[_i];
      delete this.variables[name];
    }
    return argument;
  }
};

MiyoFilters.variables_temporary_set = {
  type: 'through',
  filter: function(argument, request, id, stash) {
    var name, pname, value, _ref;
    if ((argument != null ? argument.variables_temporary_set : void 0) == null) {
      throw 'argument.variables_temporary_set undefined';
    }
    _ref = argument.variables_temporary_set;
    for (name in _ref) {
      value = _ref[name];
      pname = name.replace(/\.[^.]+$/);
      this.variables_temporary[name] = this.property(argument.variables_temporary_set, pname, request, id, stash);
    }
    return argument;
  }
};

MiyoFilters.variables_temporary_delete = {
  type: 'through',
  filter: function(argument, request, id, stash) {
    var name, _i, _len, _ref;
    if ((argument != null ? argument.variables_temporary_delete : void 0) == null) {
      throw 'argument.variables_temporary_delete undefined';
    }
    _ref = argument.variables_temporary_delete;
    for (_i = 0, _len = _ref.length; _i < _len; _i++) {
      name = _ref[_i];
      delete this.variables_temporary[name];
    }
    return argument;
  }
};

if ((typeof module !== "undefined" && module !== null) && (module.exports != null)) {
  module.exports = MiyoFilters;
}
